src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp

```diff
@@ -334,6 +334,55 @@ namespace UiaOperationAbstractionTests
             CacheRequestNavigationMethodTest(true);
         }
 
+        // This test aims to test the wrapper implementation could
+        // elegantly handle the case that UIA navigation hits a
+        // boundary element which means the navigation API does not
+        // find the desired element. In this case, we should simply
+        // return an empty element even a cache request is provided.
+        TEST_METHOD(CacheRequestNavigationNullReturnRemoteTest)
```

Comment:
>CacheRequestNavigationNullReturnRemoteTest [](start = 20, length = 42)

Is there a reason to not follow the local-remote test pairing?

Reply:
yes, the reason is `GetParentElement` will cross process boundary using local API, it is hard to know how many elements will be in the array once we step out that boundary and what is the last element. Technically, the last element should be the "desktop" node, but I am not sure if that is true since I assume this test should be able to run everywhere.

Reply:
Hmm... That is true. I think it would be great to just capture that part in the comment to justify not writing two tests for it. Because that is the existing and designed discrepancy between the two methods.

Reply:
A shorter comment would have worked, too, but this is fine. :)

---
In reply to: [509727632](https://github.com/microsoft/Microsoft-UI-UIAutomation/pull/48#discussion_r509727632) [](ancestors = 509727632)